src/dev/flang/be/interpreter/Intrinsics.java

```diff
@@ -313,6 +331,76 @@ public static Callable call(Interpreter interpreter, Clazz innerClazz)
               return new boolValue(false);
             }
         });
+    put("fuzion.std.fileio.on_open", (interpreter, innerClazz) -> args ->
+        {
+          if (!ENABLE_UNSAFE_INTRINSICS)
+            {
+              System.err.println("*** error: unsafe feature "+innerClazz+" disabled");
+              System.exit(1);
+            }
+          var open_results = (long[])args.get(2).arrayData()._array; 
+          long fd;
+          try
+            {
+              switch (args.get(3).i8Value()) {
+                case 0:
+                  InputStream fis = new FileInputStream(utf8ByteArrayDataToString(args.get(1)));
+                  // to prevent duplicate FDs
+                  fd = (_openOutputStreams_.isEmpty() && _openInputStreams_.isEmpty())? 10000 : _openOutputStreams_.size() + _openInputStreams_.size() + 10000;
+                  _openInputStreams_.put(fd, fis);
+                  open_results[0] = fd;
+                  break;
+                case 1:
+                  OutputStream fos = new FileOutputStream(utf8ByteArrayDataToString(args.get(1)));
+                  fd = (_openOutputStreams_.isEmpty() && _openInputStreams_.isEmpty())? 10000 : _openOutputStreams_.size() + _openInputStreams_.size() + 10000;
+                  _openOutputStreams_.put(fd, fos);
+                  open_results[0] = fd;
+                  break;
+                case 2:
+                  OutputStream fas = new FileOutputStream(utf8ByteArrayDataToString(args.get(1)), true);
+                  fd = (_openOutputStreams_.isEmpty() && _openInputStreams_.isEmpty())? 10000 : _openOutputStreams_.size() + _openInputStreams_.size() + 10000;
+                  _openOutputStreams_.put(fd, fas);
+                  open_results[0] = fd;
+                  break;
+                default:
+                  open_results[1] = -1;
+                  System.err.println("*** Unsupported open flag. Please use: 0 for READ, 1 for WRITE, 2 for APPEND. ***");
+                  System.exit(1);
```

Comment:
Currently in the interpreter I only return -1 in case of any errors until I find a way to return error numbers in Java that are comparable to the ones generated by the C back-end.

Reply:
some here, an internal argument with a wrong value can crash badly.